parser/src/main/scala/aqua/parser/lexer/ValueToken.scala

```diff
@@ -91,6 +94,30 @@ object CallArrowToken {
     }
 }
 
+case class DataValueToken[F[_]: Comonad](
+  typeName: CustomTypeToken[F],
+  fields: NonEmptyMap[String, ValueToken[F]]
+) extends ValueToken[F] {
+
+  override def mapK[K[_]: Comonad](fk: F ~> K): DataValueToken[K] =
+    copy(typeName.mapK(fk), fields.map(_.mapK(fk)))
+
+  override def as[T](v: T): F[T] = typeName.as(v)
+}
+
+object DataValueToken {
+
+  val dataValue: P[DataValueToken[Span.S]] =
+      (`Class`.lift
+        ~ comma(((`name` <* (` `.?.with1 *> `=` *> ` `.?)).with1 ~ ValueToken.`value`).surroundedBy(`/s*`))
+          .between(` `.?.with1 *> `(` <* `/s*`, `/s*` *> `)`))
+        .withContext(
+          "Missing braces '()' after the object name"
```

Comment:
Is it the object name? Or a struct type? From the user perspective, I suppose, it's a struct type